lib/kino/audio.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Kino.Audio do
+  @moduledoc """
+  A kino for rendering a binary audio.
+
+  ## Examples
+
+      content = File.read!("/path/to/audio.wav")
+      Kino.Audio.new(content, :wav)
+
+      content = File.read!("/path/to/audio.wav")
+      Kino.Audio.new(content, :wav, [:controls, :autoplay, :loop])
+  """
+
+  use Kino.JS
+  use Kino.JS.Live
+
+  @type t :: Kino.JS.Live.t()
+
+  @type mime_type :: binary()
+  @type common_audio_type :: :wav | :mp3 | :mpeg | :ogg
+  @doc """
+  Creates a new kino displaying the given binary audio.
+
+  The given type be either `:wav`, `:mp3`/`:mpeg`, `:ogg`
+  or a string with audio MIME type.
+
+  """
+  @spec new(binary(), common_audio_type() | mime_type(), list()) :: t()
+  def new(content, type, opts \\ [:controls, :autoplay]) when is_binary(content) do
+    Kino.JS.Live.new(__MODULE__, %{
+      content: content,
+      type: mime_type!(type),
+      opts: Enum.join(opts, " ")
+    })
+  end
+
+  @impl true
+  def init(assigns, ctx) do
+    {:ok, assign(ctx, assigns)}
+  end
+
+  @impl true
+  def handle_connect(%{assigns: %{content: content, type: type, opts: opts}} = ctx) do
+    payload = {:binary, %{type: type, opts: opts}, content}
+    {:ok, payload, ctx}
+  end
+
+  defp mime_type!(:wav), do: "audio/wav"
+  defp mime_type!(:mp3), do: "audio/mpeg"
+  defp mime_type!(:mpeg), do: "audio/mpeg"
+  defp mime_type!(:ogg), do: "audio/ogg"
+  defp mime_type!("audio/" <> _ = mime_type), do: mime_type
+
+  defp mime_type!(other) do
+    raise ArgumentError,
+          "expected audio type to be either :wav, :mp3, :mpeg, :ogg, or an audio MIME type string, got: #{inspect(other)}"
+  end
+
+  asset "main.js" do
+    """
+    export function init(ctx, [{ type, opts }, content]) {
+      let blob = new Blob([content], {type: type})
+      let data = URL.createObjectURL(blob)
```

Comment:
I'd rather use the `bufferToBase64` function I sent. I'm don't remember exactly, but I'm pretty sure there is a reason we don't convert using Blob in other places, it may have been slower. Other than Blobs should be revoked (perhaps less of an issue with iframes).

Reply:
It's ok to duplicate this function?

Reply:
Yeah it's fine :)

Reply:
Oh, also please define the assets in files, as we do here: https://github.com/livebook-dev/kino/blob/70fc3ad2ad8350789185d9dc7120812b7aa6fd45/lib/kino/html.ex#L32

It's fine readability-wise, but since recently Livebook may try to load assets from CDN, but that's only possible if they are defined in separate files :)